### ISSUE TITLE
The validation metrics score (in training step) is not identical to the evaluate function scores?

### ISSUE THREAD

**Issue by bifeng** [NTMC-Community/MatchZoo#595]
I am running the tutorials of wikiqa matchpyramid.ipynb,  and the final metrics score in validation is:

> Validation: normalized_discounted_cumulative_gain@3(0.0): 0.5703992533432362 - normalized_discounted_cumulative_gain@5(0.0): 0.6361247299004382 - mean_average_precision(0.0): 0.5847498909041217

But, i calculate the metrics score using the same validation with:
`model.evaluate(pred_x, pred_y)`

I get the scores:

> normalized_discounted_cumulative_gain@3(0.0): 0.3436525884549766, normalized_discounted_cumulative_gain@5(0.0): 0.43721582791389524, mean_average_precision(0.0): 0.39102972601138863

Why is so much difference between each other? Is that correct?

**Comment by uduse** [comment]
`EvaluateAllMetrics` calls the model's `evaluate`. Make sure you're passing the same data and make sure you don't change the data. If the problem persists, please provide a Minimal, Complete, and Verifiable example.

**Comment by bifeng** [comment]
Yes, i am running two times (running with the whole script, so there is no chance for change the data),  this problem is still exist.

**Comment by bwanglzu** [comment]
@bifeng did you trained a new model?

**Comment by uduse** [comment]
![image](https://user-images.githubusercontent.com/4717005/51605899-51639600-1f4b-11e9-9586-87ee811108df.png)
This is what I got. I got an error if I don't set `batch_size`.

**Comment by bifeng** [comment]
Thanks very much! I didn't get error, but if i set batch_size, the score is correct. So, previous result is just using the default batch_size(128). Thanks, It's my carelessness.
